### PR TITLE
add setup nix gati

### PR DESCRIPTION
**Pull request by tt-cll**

### CODE REVIEW

**Comment by tt-cll** [comment]
Pulled from https://github.com/smartcontractkit/chainlink-evm-testing/blob/main/.github/actions/nix-gati-setup/action.yml so we can re-use it in chain maintenance and other repos

File: actions/setup-nix-gati/action.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          "ssh://git@github.com/smartcontractkit/"
+
+    - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c #v26
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+        extra_nix_config: |
+          sandbox = true # force sandbox for all OS (normally disabled for macOS)
+
+    - uses: DeterminateSystems/magic-nix-cache-action@87e8236f46702ab0ce5a058b605a173ec88d618e #v6
+
```
[the remark this answers is not in the document]
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml]: good suggestion! I added a bool param to toggle it

**Review by willyrgf — APPROVED** [review]
LGTM

**Comment by tt-cll** [comment]
This worked integrated here https://github.com/smartcontractkit/chainlink-evm-testing/pull/62

This will not require a release

File: actions/setup-nix-gati/action.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Setup GitHub token
+      id: setup-github-token
+      uses: smartcontractkit/.github/actions/setup-github-token@main
```
Comment (erikburt) [inline actions/setup-nix-gati/action.yml:43]: This should either be:
1. tagged to a commit (not `@main`) or
2. referenced as local action (after checking out this repo)

File: actions/setup-nix-gati/action.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          sandbox = true # force sandbox for all OS (normally disabled for macOS)
+
+    - name: use magic nix cache
+      if: inputs.enable-magic-cache == 'true'
+      uses: DeterminateSystems/magic-nix-cache-action@87e8236f46702ab0ce5a058b605a173ec88d618e #v6
+
+    - name: Collect metrics
+      if: always()
+      id: collect-gha-metrics
+      uses: smartcontractkit/push-gha-metrics-action@0281b09807758be1dcc41651e44e62b353808c47 # v2.1.0
```
Comment (erikburt) [inline actions/setup-nix-gati/action.yml:71]: Outdated dependency

https://github.com/smartcontractkit/push-gha-metrics-action/releases

File: actions/setup-nix-gati/action.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c #v26
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+        extra_nix_config: |
+          sandbox = true # force sandbox for all OS (normally disabled for macOS)
+
+    - name: use magic nix cache
+      if: inputs.enable-magic-cache == 'true'
+      uses: DeterminateSystems/magic-nix-cache-action@87e8236f46702ab0ce5a058b605a173ec88d618e #v6
```
Comment (erikburt) [inline actions/setup-nix-gati/action.yml:66]: I see v7 was just released, should this be bumped?

https://github.com/DeterminateSystems/magic-nix-cache-action/releases

File: actions/setup-nix-gati/action.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    - name: Collect metrics
+      if: always()
```
Comment (erikburt) [inline actions/setup-nix-gati/action.yml:69]: This should be conditional on `metrics-job-name` being passed.

See this as an example diff: https://github.com/smartcontractkit/.github/pull/421/files

File: actions/setup-nix-gati/action.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    - name: Configure GitHub token from GATI
+      shell: bash
+      run: |
+        # redirects ssh url to https with the token
+        git config --global \
+          url."https://x-access-token:${{ steps.setup-github-token.outputs.access-token }}@github.com/smartcontractkit/".insteadOf \
+          "ssh://git@github.com/smartcontractkit/"
+
+    - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c #v26
```
Comment (erikburt) [inline actions/setup-nix-gati/action.yml:58]: Should this be using v27?

https://github.com/cachix/install-nix-action/releases

**Review by erikburt — CHANGES_REQUESTED** [review]

File: actions/setup-nix-gati/action.yml (hunk 3 above)
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml:71]: This was generated from the template action outlined in the wiki. Can the action generation be updated in a separate PR that has the correct dependency? https://github.com/smartcontractkit/.github?tab=readme-ov-file#creating-a-new-action

File: actions/setup-nix-gati/action.yml (hunk 5 above)
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml:69]: I can make the fix here, and I think we also need to update the action generation to have this fix.

File: actions/setup-nix-gati/action.yml (hunk 2 above)
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml:43]: good catch!

File: actions/setup-nix-gati/action.yml (hunk 4 above)
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml:66]: this is a lift and shift from https://github.com/smartcontractkit/chainlink-evm-testing/blob/main/.github/actions/nix-gati-setup/action.yml. I don't think we need to upgrade this but it also shouldn't hurt

File: actions/setup-nix-gati/action.yml (hunk 6 above)
Comment (tt-cll) [reply actions/setup-nix-gati/action.yml]: same comment re lift and shift upgrades. Also shouldn't hurt though.

File: actions/setup-nix-gati/action.yml (hunk 3 above)
Comment (erikburt) [reply actions/setup-nix-gati/action.yml]: Yes good call - I will update that.

File: actions/setup-nix-gati/action.yml (hunk 5 above)
Comment (erikburt) [reply actions/setup-nix-gati/action.yml]: Yes good catch - I will do that.

**Review by erikburt — APPROVED** [review]

**Pull request merged**